tests/fft_test.py

```diff
@@ -54,43 +56,83 @@ def _get_fftn_test_axes(shape):
   return axes
 
 
+def _get_fftn_func(module, inverse, real):
+  if inverse:
+    return _irfft_with_zeroed_inputs(module.irfftn) if real else module.ifftn
+  else:
+    return module.rfftn if real else module.fftn
+
+
+def _irfft_with_zeroed_inputs(irfft_fun):
+  # irfft isn't defined on the full domain of inputs, so in order to have a
+  # well defined derivative on the whole domain of the function, we zero-out
+  # the imaginary part of the first and possibly the last elements.
```

Comment:
This logic is pretty complicated for a test... would be it be possible/simpler to limit the test inputs to irfft?

Reply:
The challenge here is that in order for the finite difference calculation to be correct, I need to apply the restriction to the perturbation as well. I don't think our `check_grads` machinery is set-up to do that unless we modify the function passed into it.